### PR TITLE
[WIP] Further explain `env_var_name` parameter type

### PR DESCRIPTION
**Pull request by rosieyohannan**
- clear up what's going on with env_var_name param type -  original request: https://circleci.slack.com/archives/C0KBNJGRH/p1595872682016600
- move parameters section to the start of the doc as this needs to be understood for all other sections
- update a couple of examples

### CODE REVIEW

File: jekyll/_cci2/reusing-config.md [hunk 1]
```diff
@@ -627,7 +650,7 @@ steps:
 
 #### Environment Variable Name
 
-The environment variable name (``env_var_name``) parameter is a string that must match a POSIX_NAME regexp (for example, there can be no spaces or special characters). The `env_var_name` parameter is a more meaningful parameter type that enables additional checks to be performed. See [Using Environment Variables]({{ site.baseurl }}/2.0/env-vars/) for details.
+The environment variable name (`env_var_name`) parameter is a string that must match a POSIX_NAME regexp (for example, there can be no spaces or special characters). The `env_var_name` parameter is a more meaningful parameter type that enables CircleCI to check that the string that has been passed can be used as an environment variable name. For more information on using environment variables, the the guide to [Using Environment Variables]({{ site.baseurl }}/2.0/env-vars/).
```
Comment (rosieyohannan) [inline jekyll/_cci2/reusing-config.md:653]:
```suggestion
The environment variable name (`env_var_name`) parameter is a string that must match a POSIX_NAME regexp (for example, there can be no spaces or special characters). The `env_var_name` parameter is a more meaningful parameter type that enables CircleCI to check that the string that has been passed can be used as an environment variable name. For more information on environment variables, see the guide to [Using Environment Variables]({{ site.baseurl }}/2.0/env-vars/).
```

File: jekyll/_cci2/reusing-config.md [hunk 2]
```diff
@@ -7,89 +7,83 @@ categories: [configuration]
...
 ---
 
-This reusable config reference page describes how to version your [.circleci/config.yml]({{ site.baseurl }}/2.0/configuration-reference/) file and get started with reusable commands, jobs, executors and orbs.
+This guide describes how to get started with reusable commands, jobs, executors and orbs.
 
 * TOC
 {:toc}
 
-## Getting Started with Config Reuse
+## Some Notes on Reusable Configuration
```
Comment (rosieyohannan) [inline jekyll/_cci2/reusing-config.md:15]:
```suggestion
## Notes on Reusable Configuration
```

File: jekyll/_cci2/reusing-config.md [hunk 3]
```diff
@@ -7,89 +7,83 @@ categories: [configuration]
...
-
-```yaml
-version: 2.1
 jobs:
   myjob:
     docker:
       - image: "cimg/base:stable"
     steps:
-      - sayhello:
+      - sayhello: // invoke command "sayhello"
```
Comment (KyleTryon) [inline jekyll/_cci2/reusing-config.md:78]: Yaml comment is "#", not "//"

Comment (rosieyohannan) [reply jekyll/_cci2/reusing-config.md:78]: Oh yeah, thanks!

Comment (rosieyohannan) [inline jekyll/_cci2/reusing-config.md:78]:
```suggestion
      - sayhello: # invoke command "sayhello"
```

File: jekyll/_cci2/reusing-config.md [hunk 4]
```diff
@@ -193,9 +195,7 @@ jobs:
       - run: echo outside the executor
 ```
 
-**Note:** Reusable `executor` declarations are available in configuration version 2.1 and later.
-
-## **`executors`**
+### The `executors` Key
```
Comment (teesloane) [inline jekyll/_cci2/reusing-config.md:198]: I believe this change is causing our html proofer to break as `jekyll/_site/docs/2.0/orbs-faq/index.html` links to `/docs/2.0/reusing-config/#executors`

Comment (rosieyohannan) [reply jekyll/_cci2/reusing-config.md:198]: Ah yes thank you!

**Comment by teesloane** [comment]
Nice work - tons of good examples.

**Pull request merged**